mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -115,7 +116,16 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
         }
         predictUDF(col(getFeaturesCol))
       }
-      outputData = outputData.withColumn(getPredictionCol, predUDF)
+      // determine number of classes either from metadata if provided.
+      val labelSchema = dataset.schema($(labelCol))
+      // extract label metadata from label column if present, or create a nominal attribute
+      // to output the number of labels
+      val labelAttribute = Attribute.fromStructField(labelSchema) match {
```

Comment:
Reply:
1, I agree that it's better to extract labelAttribute during model fitting, but it need to be transferred to model as an argument of model construction which may take API breaking change. Is this reasonable? Actually [```OneVsRestModel```](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala#L136) has already done similar work.
2, I referred that [```OneVsRest```](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala#L347) and found the attribute was named as ```label```. May be we should rename it as well.
3, Agree.